portal-wire-protocol.md

```diff
@@ -0,0 +1,163 @@
+# Portal Wire Protocol
+
+The Portal wire protocol is the base p2p protocol by which nodes communicate.
+
+The different Portal networks **MAY** use this base protocol, but they **MUST** remain separated per network.
+This is done at the [Node Discovery Protocol v5](https://github.com/ethereum/devp2p/blob/master/discv5/discv5-wire.md#talkreq-request-0x05) layer, by providing a different protocol bytestring, per network, in the `TALKREQ` message.
+
+The protocol bytestring in the `TALKREQ` message **MUST** have "portal-" as prefix then followed by the network name (e.g. "portal-state").
+
+Each network can specify which specific messages from the protocol it supports.
+
+## Messages
+
+All messages in the protocol are transmitted using the `TALKREQ` and `TALKRESP` messages from the base protocol.
+
+All messages have a `message_id` and `encoded_message` that are concatenated to form the `payload` for either a `TALKREQ` or `TALKRESP` message.
+
+```
+payload         := message_id | encoded_message
+message_id      := uint8
+encoded_message := bytes
+```
+
+The `encoded_message` component is the SSZ encoded payload for the message as is indicated by its `message_id`. Each message has its own `sedes` which dictates how it should be encoded and decoded.
+
+The SSZ sedes `ByteList` is used to alias `List[uint8, max_length=2048]`.
+
+All messages have a `type` which is either `request` or `response`.
+
+* `request` messages **MUST** be sent using a `TALKREQ`
+* `response` messages **MUST** be sent using a `TALKRESP`
+
+
+### Ping (0x01)
+
+Request message to check if a node is reachable, communicate basic information about our node, and request basic information about the recipient node.
+
+```
+message_id := 0x01
+type       := request
+sedes      := Container(enr_seq: uint64, data_radius: uint256, custom_payload: ByteList)
+```
+
+* `enr_seq`: The node's current sequence number of their ENR record.
+* `data_radius`: The nodes current maximum radius for data stored by this node.
```

Comment:
Worth noting that the History Gossip network won't use this value at all (the way it is currently designed).  Might be worth just embedding this directly into the `custom_payload` field.

Reply:
Right, good point! Had not taken to much of gossip into account (yet). Will adjust